gcp-client/src/main/java/com/google/cloud/graphite/platforms/plugin/client/ContainerAnalysisClient.java

```diff
@@ -0,0 +1,189 @@
+import static com.google.cloud.graphite.platforms.plugin.client.util.ClientUtil.nameFromSelfLink;
+import static com.google.cloud.graphite.platforms.plugin.client.util.ClientUtil.processResourceList;
+
+import com.google.api.services.containeranalysis.v1beta1.model.Occurrence;
+import com.google.common.base.Preconditions;
+import com.google.common.base.Strings;
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.ImmutableMap;
+import java.io.IOException;
+import java.util.Comparator;
+import java.util.List;
+import java.util.Optional;
+import java.util.concurrent.TimeUnit;
+import lombok.extern.java.Log;
+import org.awaitility.Awaitility;
+import org.awaitility.core.ConditionTimeoutException;
+
+/**
+ * Client for communicating with the Container Analysis API.
+ *
+ * @see <a href="https://cloud.google.com/container-registry/docs/container-analysis">Container
+ *     Analysis</a>
+ */
+@Log
+public class ContainerAnalysisClient {
+  private static final String VULNERABILITY_NOTE_PROJECT_ID = "goog-analysis";
+  private static final String VULNERABILITY_NOTE_ID = "PACKAGE_VULNERABILITY";
+  private static final String VULNERABILITY_KIND = "VULNERABILITY";
+  private static final List<String> FINISHED_STATUSES =
+      ImmutableList.of("FINISHED_SUCCESS", "FINISHED_FAILED", "FINISHED_UNSUPPORTED");
+  private ContainerAnalysisWrapper containerAnalysis;
+
+  ContainerAnalysisClient(ContainerAnalysisWrapper containerAnalysis) {
+    this.containerAnalysis = containerAnalysis;
+  }
+
+  /**
+   * Wait until the vulnerability scan of the provided container image is complete and return the
+   * final status.
+   *
+   * @param projectId The ID of the project where the container image is hosted.
+   * @param resourceUrl The complete URL of the image in this form:
+   *     https://[HOST_NAME]/[PROJECT_ID]/[IMAGE_ID]@sha256:[HASH]
+   * @param timeoutMillis The number of milliseconds to wait for the scan before timing out.
+   * @return The final status of the scan, one of FINISHED_SUCCESS, FINISHED_FAILED, or
+   *     FINISHED_UNSUPPORTED.
+   * @throws ConditionTimeoutException If the the analysis was not in a finished state before the
+   *     timeout expired.
+   */
+  public String getVulnerabilityScanStatusSync(
+      final String projectId, final String resourceUrl, final long timeoutMillis)
+      throws ConditionTimeoutException {
+    Preconditions.checkArgument(!Strings.isNullOrEmpty(projectId));
+    Preconditions.checkArgument(!Strings.isNullOrEmpty(resourceUrl));
+    Preconditions.checkArgument(timeoutMillis >= 0);
+    long start = System.currentTimeMillis();
+    Occurrence statusOccurence =
+        Awaitility.await()
+            .timeout(timeoutMillis, TimeUnit.MILLISECONDS)
+            .until(() -> getStatusOccurrence(projectId, resourceUrl), Optional::isPresent)
+            .get();
+    long timeLeft = timeoutMillis - (System.currentTimeMillis() - start);
```

Comment:
What's the behavior when timeLeft is <= 0?

Reply:
Good catch. This would have been an IllegalArgumentException when passed to the "timeout" method, added a manual check and ConditionTimeoutException for this case. I expect this would be rare, as it's an extremely narrow time window, but better to be defensive.